node/src/accountant/scanners.rs

```diff
@@ -109,16 +116,21 @@ impl ScannerCommon {
         }
     }
 
-    fn remove_timestamp(&mut self, scan_type: ScanType, logger: &Logger) {
+    fn remove_timestamp(&mut self, scan_type: ScanType, now: SystemTime, logger: &Logger) {
```

Comment:
The thing that pricked up my ears about this method name is the fact that it was called `remove_timestamp()` and you're providing it with a timestamp.

Now that you've described its function to me, I think I'd like to see it given a higher-level name: not something about removing a timestamp, but something about why the timestamp should be removed. Maybe `signal_scanner_completion` or something along those lines?